high_card.rb

```diff
@@ -1,5 +1,117 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
-players = []
+$winning_card = nil
+$second_winning_card = nil
+$players = []
+
+$r_ordered = [2,3,4,5,6,7,8,9,10,"J","Q","K","A"]
+# 0 to 12
+$s_ordered = ["clubs", "diamonds", "hearts", "spades"]
+# 0 to 3
+
+def shuffled_deck
+
+  new_deck = []
+  $r_ordered.each_with_index do |rank, index|
+    $s_ordered.each_with_index do |suit, index_two|
+      new_deck.push({
+        rank_n: index,
+        rank: rank,
+        suit_n: index_two,
+        suit: suit
+        })
+      end
+    end
+
+    shuffled = new_deck.shuffle
+    return shuffled
+  end
+
+def collect_players # fix this
+  number_of_players = 0
+  input = ""
+  player_array = []
+  while input != "play"
+    puts "#{number_of_players} so far. Enter a player name, or type 'play'."
+    input = gets.chomp.to_s.downcase
+    if input == "play"
+      break
+    end
+    player_array.push(input)
+    number_of_players += 1
+  end
+  return player_array
+end
+
+def assign_cards_to_players
+  s_deck = shuffled_deck # array of cards
+  player_array = collect_players # array of players
+  # puts player_array.length
+  players_with_cards = []
+  require 'set'
+  my_set = Set.new
```

Comment:
I don't use the Fisher-Yates shuffle algorithm to shuffle.  My way is not
as efficient since I use a set as a filter but shuffles a bit more.

-Nathan

On Sat, Oct 31, 2015 at 4:41 PM, Nicholas Olds <notifications@github.com>
wrote:

> In high_card.rb
> <https://github.com/ga-dc/high_card/pull/75#discussion_r43575976>:
>
> > +    if input == "play"
> > +      break
> > +    end
> > +    player_array.push(input)
> > +    number_of_players += 1
> > +  end
> > +  return player_array
> > +end
> > +
> > +def assign_cards_to_players
> > +  s_deck = shuffled_deck # array of cards
> > +  player_array = collect_players # array of players
> > +  # puts player_array.length
> > +  players_with_cards = []
> > +  require 'set'
> > +  my_set = Set.new
>
> What is the specific use case of a set here in comparison to the
> random_array you declare below
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga-dc/high_card/pull/75/files#r43575976>.
>